### PR TITLE
Bump mkdocs and all deps to most recent versions

### PR DESCRIPTION
**Pull request by suricactus**
I have `pip freezeed` all deps, IMO it is better to have a more precise snapshot on all the deps rather than just some of them.

### CODE REVIEW

**Comment by m-kuhn** [comment]
@why-not-try-calmer dare having a look?

**Comment by why-not-try-calmer** [comment]
On it

File: documentation/index.md [hunk 1]
```diff
@@ -2,3 +2,5 @@
 template: home.html
 title: QField Ecosystem Documentation
 ---
+
+# QField Ecosystem Documentation
```
Comment (suricactus) [inline documentation/index.md:6]: NOTE if this is an empty file, it blows up for unknown reason. Decided to reuse a string that is already translated as a workaround.

File: mkdocs.yml [hunk 2]
```diff
@@ -124,8 +124,8 @@ markdown_extensions:
       alternate_style: true
   - fancyboxmd
   - pymdownx.emoji:
-      emoji_index: !!python/name:materialx.emoji.twemoji
-      emoji_generator: !!python/name:materialx.emoji.to_svg
+      emoji_index: !!python/name:material.extensions.emoji.twemoji
+      emoji_generator: !!python/name:material.extensions.emoji.to_svg
```
Comment (suricactus) [inline mkdocs.yml:128]: The `materialx` is deprecated and this fixes a warning.

https://squidfunk.github.io/mkdocs-material/setup/extensions/#recommended-configuration

**Comment by why-not-try-calmer** [comment]
> I have `pip freezeed` all deps, IMO it is better to have a more precise snapshot on all the deps rather than just some of them.

pre-commit was not freezed. Anything else?

**Comment by suricactus** [comment]
I did not freeze the dev dependencies, IMO we should not do that?

**Comment by why-not-try-calmer** [comment]
Beside that it builds fine, 

> I did not freeze the dev dependencies, IMO we should not do that?

Who which dependencies where frozen?

**Comment by why-not-try-calmer** [comment]
Oh I see you mean that you computed all transitive dependencies from the dependencies enumerated in requirements and pinned them. I believe this make the build step depend on our pins instead of the versions preferred by the top-level (non-transitive) dependencies. What happens if, for example, the maintainers of mkdocs want to change the version of click from A to B, where A is the one you pinned and B is the one they prefer?

**Comment by suricactus** [comment]
Yes, this is correct. I just used `pip freeze`. There are pros and cons in both cases, I do prefer having all dependencies pinned as it guarantees better reproducability, but it is more a matter of taste.

**Comment by why-not-try-calmer** [comment]
> Yes, this is correct. I just used `pip freeze`. There are pros and cons in both cases, I do prefer having all dependencies pinned as it guarantees better reproducability, but it is more a matter of taste.

But if mkdocs switches trans. dep 1 from 1A to 1B and we pin 1A, are we not guaranteeing the reproducibility of a _broken_ build?

> Yes, this is correct. I just used `pip freeze`. There are pros and cons in both cases, I do prefer having all dependencies pinned as it guarantees better reproducability, but it is more a matter of taste.

> Yes, this is correct. I just used `pip freeze`. There are pros and cons in both cases, I do prefer having all dependencies pinned as it guarantees better reproducability, but it is more a matter of taste.

But if mkdocs switches trans. dep 1 from 1A to 1B and we pin 1A, are we not guaranteeing the reproducibility of a _broken_ build?

**Comment by suricactus** [comment]
> But if mkdocs switches trans. dep 1 from 1A to 1B and we pin 1A, are we not guaranteeing the reproducibility of a broken build? 

If `mkdocs` changes dep `1A` to `1B`, we need to change the version of `mkdocs` too. If they require dep to be `1B`, then our `requirements.txt` will fail to install, as we said `==1A`, but `mkdocs` requires `==1B` and we should react.

Anyways, both has pros and cons, I put only the direct dependencies in `requirements.txt` and we can leave the discussion about pros and cons for better times.

**Review by why-not-try-calmer — APPROVED** [review]
I much prefer without the transitive dependencies for this particular use case, thanks!

**Comment by suricactus** [comment]
Thanks for the review and green light! Let's hope it all build correctly!

**Comment by why-not-try-calmer** [comment]
> Thanks for the review and green light! Let's hope it all build correctly!

I built it on my machine. Should I add a build step to the tests?

**Pull request merged**